### ISSUE TITLE
CEServer - How to use mono features?

### ISSUE THREAD

**Issue by Symbai** [cheat-engine/cheat-engine#1144]
How do I use mono features on a unity game running on Linux?

**Comment by cheat-engine** [comment]
you don't it's not supported right now

**Comment by Symbai** [comment]
Is it enough to compile the mono collector dll for Linux and inject it or are unity games on Linux completely different?

**Comment by cheat-engine** [comment]
No, you need to change the way the modules are scanned inside the target process
You need to setup a completely new communication method  (perhaps sockets instead of pipes)
and rewrite the lua code of the mono plugin to deal with the new method

**Comment by ShadiestGoat** [comment]
Hey is there any update on this?

**Comment by cheat-engine** [comment]
slightly.  The mac version has it configured to use sockets, but besides that nothing specific for linux/android yet

**Comment by cheat-engine** [comment]
yes it works

**Comment by cheat-engine** [comment]
if you use an emulator it's usually x86 not arm.

anyhow you also need to build the extension and libmonodatacollector  (the eclipse cproject file is at https://github.com/cheat-engine/cheat-engine/tree/master/Cheat%20Engine/MonoDataCollector )
(i'm not sure why it'd for linux-arm and not just arm)

**Comment by cheat-engine** [comment]
that means that the arm code then goes through the arm to x86 recompiler and all the code will be x86 including the registers and everything else.
